services/QuillLMS/app/lib/rule_feedback_history.rb

```diff
@@ -48,14 +48,18 @@ def self.feedback_history_to_json(f_h)
         entry: f_h.entry,
         highlight: f_h.metadata.instance_of?(Hash) ? f_h.metadata['highlight'] : '',
         session_uid: f_h.feedback_session_uid,
-        strength: f_h.feedback_history_ratings.order(updated_at: :desc).first&.rating
+        strength: f_h.feedback_history_ratings.sort_by(&:updated_at).last&.rating
```

Comment:
Gotcha: I observed empirically that calling `order` on an ActiveRecord::Relation triggers an n+1 query here, even when using `includes` in `generate_rulewise_report`. Sorting in memory does not trigger the n+1 query.